fc-community-backend/apps/api/src/factchain-core/web3.ts

```diff
@@ -74,37 +69,35 @@ export class FactChainBackend implements NoteReader, NoteWriter {
     };
   };
 
-  getNotes = async (postUrl: string): Promise<Array<Note>> => {
+  getNotes = async (
+    predicate: (note: Note) => boolean,
+    lookBackDays: number,
+  ): Promise<Array<Note>> => {
     const currentBlockNumber = await this._provider.getBlockNumber();
     const today = new Date();
-    const lookbackDays = parseInt(this._config.LOOKBACK_DAYS);
-    const from = new Date(today.getTime() - lookbackDays * 24 * 60 * 60 * 1000);
-    console.log(
-      `Getting notes created on '${postUrl}' between ${from} and ${today}`,
-    );
+    const from = new Date(today.getTime() - lookBackDays * 24 * 60 * 60 * 1000);
+    console.log(`getting notes between ${from} and ${today}`);
 
     const block_periods = timePeriodToBlockPeriods(
       from,
       today,
       currentBlockNumber,
     );
-    let notePromises: Promise<Note>[] = [];
+    let notes: Note[] = [];
+
     for (const period of block_periods) {
       const events = await this.getEvents("NoteCreated", period[0], period[1]);
+      const allCreatedNotes = await Promise.all(
+        events.map(async (event) => {
+          return await this.getNote(event.args[0], event.args[1]);
+        }),
```

Comment:
I see what you mean about the issue of selecting all, and only then filtering.

Maybe we can keep things simple and efficient for now by expecting a `predicate: (postUrl, creatorAddress) => boolean` ?
This way you can still do

```javascript
const relatedEvents = events.filter((e) => predicate(e.args[0], e.args[1]));
if (relatedEvents) {
  notePromises = notePromises.concat(
    relatedEvents.map(async (event) => {
      return await this.getNote(event.args[0], event.args[1]);
    }),
  );
}
...
```